### PR TITLE
[ISSUE #3255] optimize the performance of AbstractShenyuPlugin#filterSelector

### PR DESCRIPTION
**Pull request by dragon-zhang**
see #3255 , I also wrote a simple performance test in `PredicateJudgeFactoryTest#comparePerformance`.

### CODE REVIEW

File: shenyu-common/src/main/java/org/apache/shenyu/common/enums/OperatorEnum.java [hunk 1]
```diff
@@ -55,7 +55,17 @@ public enum OperatorEnum {
      * Contains operator enum.
      */
     CONTAINS("contains", true),
-    
+
+    /**
+     * Starts operator enum.
+     */
+    STARTS("starts", true),
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/enums/OperatorEnum.java:62]: modify this to start_with

File: shenyu-common/src/main/java/org/apache/shenyu/common/enums/OperatorEnum.java [hunk 2]
```diff
@@ -55,7 +55,17 @@ public enum OperatorEnum {
...
+
+    /**
+     * Ends operator enum.
+     */
+    ENDS("ends", true),
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/enums/OperatorEnum.java:67]: modify this to end_with

File: shenyu-common/src/main/java/org/apache/shenyu/common/enums/OperatorEnum.java (hunk 1 above)
Comment (dragon-zhang) [reply shenyu-common/src/main/java/org/apache/shenyu/common/enums/OperatorEnum.java:62]: > modify this to start_with

ok

File: shenyu-plugin/shenyu-plugin-base/src/test/java/org/apache/shenyu/plugin/base/condition/judge/PredicateJudgeFactoryTest.java [hunk 3]
```diff
@@ -123,4 +128,42 @@ public void testContainsJudge() {
...
+    @Test
+    public void testEndsJudge() {
+        conditionData.setOperator(OperatorEnum.ENDS_WITH.getAlias());
+        assertTrue(PredicateJudgeFactory.judge(conditionData, "/**/test/http/"));
+        assertFalse(PredicateJudgeFactory.judge(conditionData, "/test/http/**"));
+        assertFalse(PredicateJudgeFactory.judge(conditionData, "/**/http1/"));
+    }
+
+    @Test
+    public void comparePerformance() {
```
Comment (yu199195) [inline shenyu-plugin/shenyu-plugin-base/src/test/java/org/apache/shenyu/plugin/base/condition/judge/PredicateJudgeFactoryTest.java:148]: remove this

**Review by yu199195 — APPROVED** [review]

**Pull request merged**